File: Selenium/Activities/Activity7.java
package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;


import java.io.IOException;

public class Activity7 {
    public static void main(String[] args) throws IOException {
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/drag-drop");
        System.out.println("Title of webpage:"+driver.getTitle());
        Actions builder=new Actions(driver);

        // Find the football
        WebElement football = driver.findElement(By.id("draggable"));
        // Find the dropzone1
        WebElement dropzone1 = driver.findElement(By.id("droppable"));
        // Find the dropzone2
        WebElement dropzone2 = driver.findElement(By.id("dropzone2"));

        builder.clickAndHold(football).moveToElement(dropzone1).release().build().perform();
        String verifyDropped1=driver.findElement(By.xpath("//div[@id='droppable']/p")).getText();
        if(verifyDropped1.equalsIgnoreCase("Dropped!")){
            System.out.println("Ball dropped to Dropzone 1");
        }
        builder.dragAndDrop(dropzone1,dropzone2).perform();
        String verifyDropped2=driver.findElement(By.xpath("//div[@id='dropzone2']/p")).getText();
        if(verifyDropped2.equalsIgnoreCase("Dropped!")){
            System.out.println("Ball dropped to Dropzone 2");
        }
        driver.close();
    }
}

File: Selenium/Activities/Activity7.py
from selenium import webdriver
from selenium.webdriver import Keys, ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/drag-drop")
    print("Page title is: ", driver.title)
    actions = ActionChains(driver)
    footBall = driver.find_element(By.ID, "draggable")
    dropZoneOne = driver.find_element(By.ID, "droppable")
    dropZoneTwo = driver.find_element(By.ID, "dropzone2")
    actions.click_and_hold(footBall).move_to_element(dropZoneOne).release().perform()
    verifyDroppedOne = driver.find_element(By.XPATH, "//div[@id='droppable']/p").text
    if verifyDroppedOne == "Dropped!":
        print("Ball dropped to Dropzone 1")
    actions.drag_and_drop(dropZoneOne, dropZoneTwo).perform()
    verifyDroppedTwo = driver.find_element(By.XPATH, "//div[@id='dropzone2']/p").text
    if verifyDroppedOne == "Dropped!":
        print("Ball dropped to Dropzone 2")
    driver.close()
